app/backend/src/couchers/servicers/threads.py

```diff
@@ -0,0 +1,103 @@
+import logging
+
+import grpc
+from google.protobuf import empty_pb2
+from sqlalchemy.sql import func
+
+from couchers import errors
+from couchers.db import is_valid_date, session_scope
+from couchers.models import Comment, Reply, Thread
+from couchers.utils import Timestamp_from_datetime
+from pb import threads_pb2, threads_pb2_grpc
+
+logger = logging.getLogger(__name__)
+
+PAGINATION_LENGTH = 10
+
+
+# Since the API exposes a single ID space regardless of nesting level,
+# we construct the API id by appending the the nesting level to the
+# database ID.
```

Comment:
How about using a shared sequence instead? That would guarantee unique IDs across the two tables, and be less hassle from the backend?

Reply:
It will be more hassle in the backend as we can't see directly on the ID which table to select from.